src/CodeGen_C.cpp

```diff
@@ -1583,6 +1583,23 @@ void CodeGen_C::compile(const LoweredFunc &f) {
         }
         stream << "\n";
     }
+
+    if (is_header() && f.linkage == LoweredFunc::ExternalPlusMetadata) {
+        // C syntax for function-that-returns-function-pointer is fun.
+        const string getter = R"INLINE_CODE(
+#ifdef HALIDE_GET_STANDARD_ARGV_FUNCTION
```

Comment:
I think there should be a comment in the generated header that explains how to use this (define X macros before including this header to get functions named Y that do Z)